tests/series/test_temporal_ops.py

```diff
@@ -120,11 +120,70 @@ def ts_maker(h):
 
     input_ts = list(map(ts_maker, input))
     s = Series.from_pylist(input_ts).cast(DataType.timestamp(TimeUnit.ms()))
-    days = s.dt.hour()
+    hours = s.dt.hour()
 
-    assert days.datatype() == DataType.uint32()
+    assert hours.datatype() == DataType.uint32()
 
-    assert input == days.to_pylist()
+    assert input == hours.to_pylist()
+
+    s = Series.from_pylist(input_ts).cast(DataType.time(TimeUnit.ns()))
+    hours = s.dt.hour()
+
+    assert hours.datatype() == DataType.uint32()
+
+    assert input == hours.to_pylist()
+
+
+def test_series_timestamp_minute() -> None:
+    from datetime import datetime
+
+    def ts_maker(mi):
+        if mi is None:
+            return None
+        return datetime(2023, 1, 26, 23, mi, 1)
+
+    input = [1, 5, 14, None, 23, None, 21]
+
+    input_ts = list(map(ts_maker, input))
+    s = Series.from_pylist(input_ts).cast(DataType.timestamp(TimeUnit.ms()))
+    minutes = s.dt.minute()
+
+    assert minutes.datatype() == DataType.uint32()
+
+    assert input == minutes.to_pylist()
+
+    s = Series.from_pylist(input_ts).cast(DataType.time(TimeUnit.ns()))
+    minutes = s.dt.minute()
+
+    assert minutes.datatype() == DataType.uint32()
+
+    assert input == minutes.to_pylist()
```

Comment:
I would suggest making the time and timestamp type assertions into separate tests. Same for hour and second